sequencing-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -262,31 +262,122 @@ export class Sequence implements SeqJson {
             }),
           }
         : {}),
+      ...(this.locals ? { locals: this.locals } : {}),
+      ...(this.parameters ? { parameters: this.parameters } : {}),
+      ...(this.requests
+        ? {
+            requests: this.requests.map(request => {
+              return {
+                name: request.name,
+                steps: [
+                  request.steps[0] instanceof CommandStem ||
+                  request.steps[0] instanceof Ground_Block ||
+                  request.steps[0] instanceof Ground_Event
+                    ? request.steps[0].toSeqJson()
+                    : request.steps[0],
+                  // @ts-ignore : 'step' found in JSON Spec
+                  ...request.steps.slice(1).map(step => {
+                    if (step instanceof CommandStem || step instanceof Ground_Block || step instanceof Ground_Event)
+                      return step.toSeqJson();
+                    return step;
```

Comment:
I'm confused how this whole piece of code works, if the first element in the list of request steps is a CommandStem, Ground_Block, or Ground_Event then we call `toSeqJson()`, and then we look at the rest of the array `slice(1)` and do the same check again?

Can't we just run `toSeqJson()` on every element of the array if it's one of those three types?

Reply:
For the array slice, we can run `toSeqJson()` on everything.

Reply:
I agree it is confusing!!!